src/test/java/com/cloudant/tests/UnicodeTest.java

```diff
@@ -94,21 +83,26 @@ public void tearDown() {
     /**
      * Returns the charset of a plain-text entity.
      */
-    private static Charset getPlainTextEntityCharset(HttpEntity entity, URI uri) {
+    private static Charset getPlainTextEntityCharset(HttpConnection connection, URI uri) {
         // For plain text, use the charset that is mentioned in the response
         // header field 'Content-Type'.
         // See http://stackoverflow.com/questions/3216730/with-httpclient-is-there-a-way-to-get
         // -the-character-set-of-the-page-with-a-head
-        ContentType contentType;
-        try {
-            contentType = ContentType.getOrDefault(entity);
-        } catch (UnsupportedCharsetException e) {
-            throw new RuntimeException("The output of " + uri + " is in charset " + e
-                    .getCharsetName() + ", which is unsupported.",
-                    e);
+
+        String contentType = connection.getConnection().getHeaderField("Content-Type");
```

Comment:
Yes, `getContentType()` refers to the remote object so we can use that - fixed in 3b5db94.
`getContentEncoding()` does not get charset information for text types, it is used for entire body compression schmes (e.g. `gzip`) so we can't use that one.